========
cmo_gatk
========
VariantAnnotator
################
.. program-output:: cmo_gatk --cmd VariantAnnotator -h

BeagleOutputToVCF
#################
.. program-output:: cmo_gatk --cmd BeagleOutputToVCF -h

ProduceBeagleInput
##################
.. program-output:: cmo_gatk --cmd ProduceBeagleInput -h

VariantsToBeagleUnphased
########################
.. program-output:: cmo_gatk --cmd VariantsToBeagleUnphased -h

AnalyzeCovariates
#################
.. program-output:: cmo_gatk --cmd AnalyzeCovariates -h

BaseRecalibrator
################
.. program-output:: cmo_gatk --cmd BaseRecalibrator -h

CallableLoci
############
.. program-output:: cmo_gatk --cmd CallableLoci -h

CompareCallableLoci
###################
.. program-output:: cmo_gatk --cmd CompareCallableLoci -h

DepthOfCoverage
###############
.. program-output:: cmo_gatk --cmd DepthOfCoverage -h

GCContentByInterval
###################
.. program-output:: cmo_gatk --cmd GCContentByInterval -h

DiagnoseTargets
###############
.. program-output:: cmo_gatk --cmd DiagnoseTargets -h

BaseCoverageDistribution
########################
.. program-output:: cmo_gatk --cmd BaseCoverageDistribution -h

CoveredByNSamplesSites
######################
.. program-output:: cmo_gatk --cmd CoveredByNSamplesSites -h

ErrorRatePerCycle
#################
.. program-output:: cmo_gatk --cmd ErrorRatePerCycle -h

FindCoveredIntervals
####################
.. program-output:: cmo_gatk --cmd FindCoveredIntervals -h

ReadGroupProperties
###################
.. program-output:: cmo_gatk --cmd ReadGroupProperties -h

ReadLengthDistribution
######################
.. program-output:: cmo_gatk --cmd ReadLengthDistribution -h

GATKPaperGenotyper
##################
.. program-output:: cmo_gatk --cmd GATKPaperGenotyper -h

FastaAlternateReferenceMaker
############################
.. program-output:: cmo_gatk --cmd FastaAlternateReferenceMaker -h

FastaReferenceMaker
###################
.. program-output:: cmo_gatk --cmd FastaReferenceMaker -h

FastaStats
##########
.. program-output:: cmo_gatk --cmd FastaStats -h

VariantFiltration
#################
.. program-output:: cmo_gatk --cmd VariantFiltration -h

UnifiedGenotyper
################
.. program-output:: cmo_gatk --cmd UnifiedGenotyper -h

HaplotypeCaller
###############
.. program-output:: cmo_gatk --cmd HaplotypeCaller -h

HaplotypeResolver
#################
.. program-output:: cmo_gatk --cmd HaplotypeResolver -h

IndelRealigner
##############
.. program-output:: cmo_gatk --cmd IndelRealigner -h

LeftAlignIndels
###############
.. program-output:: cmo_gatk --cmd LeftAlignIndels -h

RealignerTargetCreator
######################
.. program-output:: cmo_gatk --cmd RealignerTargetCreator -h

QualifyMissingIntervals
#######################
.. program-output:: cmo_gatk --cmd QualifyMissingIntervals -h

PhaseByTransmission
###################
.. program-output:: cmo_gatk --cmd PhaseByTransmission -h

ReadBackedPhasing
#################
.. program-output:: cmo_gatk --cmd ReadBackedPhasing -h

CheckPileup
###########
.. program-output:: cmo_gatk --cmd CheckPileup -h

CountBases
##########
.. program-output:: cmo_gatk --cmd CountBases -h

CountIntervals
##############
.. program-output:: cmo_gatk --cmd CountIntervals -h

CountLoci
#########
.. program-output:: cmo_gatk --cmd CountLoci -h

CountMales
##########
.. program-output:: cmo_gatk --cmd CountMales -h

CountReadEvents
###############
.. program-output:: cmo_gatk --cmd CountReadEvents -h

CountReads
##########
.. program-output:: cmo_gatk --cmd CountReads -h

CountRODs
#########
.. program-output:: cmo_gatk --cmd CountRODs -h

CountRODsByRef
##############
.. program-output:: cmo_gatk --cmd CountRODsByRef -h

CountTerminusEvent
##################
.. program-output:: cmo_gatk --cmd CountTerminusEvent -h

ErrorThrowing
#############
.. program-output:: cmo_gatk --cmd ErrorThrowing -h

FlagStat
########
.. program-output:: cmo_gatk --cmd FlagStat -h

Pileup
######
.. program-output:: cmo_gatk --cmd Pileup -h

PrintRODs
#########
.. program-output:: cmo_gatk --cmd PrintRODs -h

QCRef
#####
.. program-output:: cmo_gatk --cmd QCRef -h

ReadClippingStats
#################
.. program-output:: cmo_gatk --cmd ReadClippingStats -h

ClipReads
#########
.. program-output:: cmo_gatk --cmd ClipReads -h

PrintReads
##########
.. program-output:: cmo_gatk --cmd PrintReads -h

ReadAdaptorTrimmer
##################
.. program-output:: cmo_gatk --cmd ReadAdaptorTrimmer -h

SplitSamFile
############
.. program-output:: cmo_gatk --cmd SplitSamFile -h

ASEReadCounter
##############
.. program-output:: cmo_gatk --cmd ASEReadCounter -h

SplitNCigarReads
################
.. program-output:: cmo_gatk --cmd SplitNCigarReads -h

SimulateReadsForVariants
########################
.. program-output:: cmo_gatk --cmd SimulateReadsForVariants -h

GenotypeAndValidate
###################
.. program-output:: cmo_gatk --cmd GenotypeAndValidate -h

Genotyper
#########
.. program-output:: cmo_gatk --cmd Genotyper -h

ValidationSiteSelector
######################
.. program-output:: cmo_gatk --cmd ValidationSiteSelector -h

VariantEval
###########
.. program-output:: cmo_gatk --cmd VariantEval -h

ApplyRecalibration
##################
.. program-output:: cmo_gatk --cmd ApplyRecalibration -h

VariantRecalibrator
###################
.. program-output:: cmo_gatk --cmd VariantRecalibrator -h

CalculateGenotypePosteriors
###########################
.. program-output:: cmo_gatk --cmd CalculateGenotypePosteriors -h

CombineGVCFs
############
.. program-output:: cmo_gatk --cmd CombineGVCFs -h

CombineVariants
###############
.. program-output:: cmo_gatk --cmd CombineVariants -h

FilterLiftedVariants
####################
.. program-output:: cmo_gatk --cmd FilterLiftedVariants -h

GenotypeConcordance
###################
.. program-output:: cmo_gatk --cmd GenotypeConcordance -h

GenotypeGVCFs
#############
.. program-output:: cmo_gatk --cmd GenotypeGVCFs -h

LeftAlignAndTrimVariants
########################
.. program-output:: cmo_gatk --cmd LeftAlignAndTrimVariants -h

LiftoverVariants
################
.. program-output:: cmo_gatk --cmd LiftoverVariants -h

RandomlySplitVariants
#####################
.. program-output:: cmo_gatk --cmd RandomlySplitVariants -h

RegenotypeVariants
##################
.. program-output:: cmo_gatk --cmd RegenotypeVariants -h

SelectHeaders
#############
.. program-output:: cmo_gatk --cmd SelectHeaders -h

SelectVariants
##############
.. program-output:: cmo_gatk --cmd SelectVariants -h

ValidateVariants
################
.. program-output:: cmo_gatk --cmd ValidateVariants -h

VariantsToAllelicPrimitives
###########################
.. program-output:: cmo_gatk --cmd VariantsToAllelicPrimitives -h

VariantsToBinaryPed
###################
.. program-output:: cmo_gatk --cmd VariantsToBinaryPed -h

VariantsToTable
###############
.. program-output:: cmo_gatk --cmd VariantsToTable -h

VariantsToVCF
#############
.. program-output:: cmo_gatk --cmd VariantsToVCF -h

VariantValidationAssessor
#########################
.. program-output:: cmo_gatk --cmd VariantValidationAssessor -h

